File: dataModule/src/main/java/com/mobnetic/coinguardian/model/market/Dashcurex.java
package com.mobnetic.coinguardian.model.market;

import java.util.HashMap;
import java.util.LinkedHashMap;

import org.json.JSONObject;

import com.mobnetic.coinguardian.model.CheckerInfo;
import com.mobnetic.coinguardian.model.Market;
import com.mobnetic.coinguardian.model.Ticker;
import com.mobnetic.coinguardian.model.currency.Currency;
import com.mobnetic.coinguardian.model.currency.VirtualCurrency;

public class Dashcurex extends Market {

	private final static String NAME = "Dashcurex";
	private final static String TTS_NAME = NAME;
	private final static String URL = "https://dashcurex.com/api/%1$s/ticker.json";
	private final static HashMap<String, CharSequence[]> CURRENCY_PAIRS = new LinkedHashMap<String, CharSequence[]>();
	static {
		CURRENCY_PAIRS.put(VirtualCurrency.DASH, new String[]{
				Currency.PLN,
				Currency.USD
			});
	}
	
	public Dashcurex() {
		super(NAME, TTS_NAME, CURRENCY_PAIRS);
	}

	@Override
	public String getUrl(int requestId, CheckerInfo checkerInfo) {
		return String.format(URL, checkerInfo.getCurrencyCounterLowerCase());
	}
	
	@Override
	protected void parseTickerFromJsonObject(int requestId, JSONObject jsonObject, Ticker ticker, CheckerInfo checkerInfo) throws Exception {
		ticker.bid = parsePrice(jsonObject.getDouble("best_ask"));
		ticker.ask = parsePrice(jsonObject.getDouble("best_bid"));
		ticker.vol = parseBTC(jsonObject.getDouble("total_volume"));
		ticker.high = parsePrice(jsonObject.getDouble("highest_tx_price"));
		ticker.low = parsePrice(jsonObject.getDouble("lowest_tx_price"));
		ticker.last = parsePrice(jsonObject.getDouble("last_tx_price"));
	}
	
	private double parsePrice(double price) {
		return price/10000;
	}
	
	private double parseBTC(double satoshi) {
		return satoshi/100000000;
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/market/Dashcurex.kt
package com.aneonex.bitcoinchecker.datamodule.model.market

import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo
import com.aneonex.bitcoinchecker.datamodule.model.Market
import com.aneonex.bitcoinchecker.datamodule.model.Ticker
import com.aneonex.bitcoinchecker.datamodule.model.currency.Currency
import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrencyPairsMap
import com.aneonex.bitcoinchecker.datamodule.model.currency.VirtualCurrency
import org.json.JSONObject

class Dashcurex : Market(NAME, TTS_NAME, CURRENCY_PAIRS) {
    companion object {
        private const val NAME = "Dashcurex"
        private const val TTS_NAME = NAME
        private const val URL = "https://dashcurex.com/api/%1\$s/ticker.json"
        private val CURRENCY_PAIRS: CurrencyPairsMap = CurrencyPairsMap()

        init {
            CURRENCY_PAIRS[VirtualCurrency.DASH] = arrayOf(
                    Currency.PLN,
                    Currency.USD
            )
        }
    }

    override fun getUrl(requestId: Int, checkerInfo: CheckerInfo): String {
        return String.format(URL, checkerInfo.currencyCounterLowerCase)
    }

    @Throws(Exception::class)
    override fun parseTickerFromJsonObject(requestId: Int, jsonObject: JSONObject, ticker: Ticker, checkerInfo: CheckerInfo) {
        ticker.bid = parsePrice(jsonObject.getDouble("best_ask"))
        ticker.ask = parsePrice(jsonObject.getDouble("best_bid"))
        ticker.vol = parseBTC(jsonObject.getDouble("total_volume"))
        ticker.high = parsePrice(jsonObject.getDouble("highest_tx_price"))
        ticker.low = parsePrice(jsonObject.getDouble("lowest_tx_price"))
        ticker.last = parsePrice(jsonObject.getDouble("last_tx_price"))
    }

    private fun parsePrice(price: Double): Double {
        return price / 10000
    }

    private fun parseBTC(satoshi: Double): Double {
        return satoshi / 100000000
    }
}